libs/entitlement/qa/common/src/main/java/org/elasticsearch/entitlement/qa/common/RestEntitlementsCheckAction.java

```diff
@@ -93,11 +125,17 @@ private static void processBuilder_startPipeline() {
         }
     }
 
+    private static void setHttpsConnectionProperties() {
+        logger.info("Calling setSSLSocketFactory");
+        var connection = new TestHttpsURLConnection();
+        connection.setSSLSocketFactory(new TestSSLSocketFactory());
```

Comment:
Would it be appropriate to assert that `connection.getSSLSocketFactory` returns the same socket factory? Or are we generally deciding that it's not the job of these tests to ensure that the protected methods actually work correctly?

Reply:
Yes, the latter: I think we want to keep these tests as short and compact as possible, focusing on passing/throwing when the method is called. I can add the check if you disagree

Reply:
No I think you're right, it's not appropriate for us to check that JCL methods work correctly. And from our knowledge of the operating principles of the system, we can have high confidence that if one of them works, they all work, since they're all instrumented the same way.